pandas/core/indexes/datetimes.py

```diff
@@ -334,7 +334,7 @@ def __new__(cls, data=None,
         if not (is_datetime64_dtype(data) or is_datetimetz(data) or
                 is_integer_dtype(data)):
             data = tools.to_datetime(data, dayfirst=dayfirst,
-                                     yearfirst=yearfirst)
+                                     yearfirst=yearfirst, cache=False)
```

Comment:
reason for this chanege?

Reply:
This is in order to prevent a `RuntimeError` due to recursion.

The cache is built using `_convert_listlike` which can return a `DatetimeIndex`, then the `DatetimeIndex` constructor can call `to_datetime` which goes back to `_convert_listlike`...